client/app/components/queries/SchemaBrowser.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { react2angular } from 'react2angular';
+import PropTypes from 'prop-types';
+import { Icon } from 'antd';
+import { debounce } from 'lodash';
+import { $rootScope } from '@/services/ng';
+
+import { Schema } from '../proptypes';
+
+const isSchemaEmpty = schema => schema === undefined || !schema.length;
+
+// Return true if the search term is contained within the table name
+// or any of the table's columns.
+const doesTableContainSearchTerm = (searchTerm, table) => {
+  const regEx = new RegExp(searchTerm, 'gi');
+
```

Comment:
One more notice about this function: avoid using regex, especially in this form. First of all, it allows users to use arbitrary regex to filter schema, as well as it breaks previous behavior. Also, using `g` modifier makes regex "sticky" - it will remember last position where match was found, and next call to `test` will start matching from that position.